### PR TITLE
[nginx-ingress] fix failover inlet

### PR DESCRIPTION
**Pull request by fl64**
## Description
Fix `HostWithFailover` inlet to work with cilium CNI.

## Why do we need it, and what problem does it solve?
`HostWithFailover` inlet doesn't work with cilium CNI.

## What is the expected result?
`HostWithFailover` inlet works as expected.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: nginx-ingress
type: fix
summary: Fix HostWithFailover inlet to work with cilium CNI.
impact: All proxy-<ingress-name>-failover daemonsets will be restarted
impact_level: default
```

### CODE REVIEW

File: modules/402-ingress-nginx/images/proxy-failover-iptables/rootfs/usr/bin/iptables-loop [hunk 1]
```diff
@@ -24,12 +24,22 @@ MARK_HTTPS_RULE="-p tcp --dport 443 -j CONNMARK --set-mark 3"
...
+DEV_NAME=ingressfailover
 # Initialization
 
+# check if interface exists
+if ! $(ip addr show "$DEV_NAME" >/dev/null 2>&1); then
+  ip link add "$DEV_NAME" type dummy
+fi
+# check if address set
+if ! $(ip addr show "$DEV_NAME" | grep -q "169.254.20.11/32"); then
+  ip addr add 169.254.20.11/32 dev "$DEV_NAME"
```
[the remark this answers is not in the document]
Comment (RomanenkoDenys) [reply modules/402-ingress-nginx/images/proxy-failover-iptables/rootfs/usr/bin/iptables-loop:39]: or may be this is must be helm variable to use it in this script and in nginx conf

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
little changes

File: modules/402-ingress-nginx/images/proxy-failover-iptables/rootfs/usr/bin/iptables-loop (hunk 1 above)
[the remark this answers is not in the document]
Comment (fl64) [reply modules/402-ingress-nginx/images/proxy-failover-iptables/rootfs/usr/bin/iptables-loop:39]: Do we need to be able to override this address, via the env in the pod?

[the remark this answers is not in the document]
Comment (RomanenkoDenys) [reply modules/402-ingress-nginx/images/proxy-failover-iptables/rootfs/usr/bin/iptables-loop]: no, it is for simplify change action

[the remark this answers is not in the document]
Comment (fl64) [reply modules/402-ingress-nginx/images/proxy-failover-iptables/rootfs/usr/bin/iptables-loop]: ok

**Review by RomanenkoDenys — APPROVED** [review]
lgtm

**Review by yalosev — APPROVED** [review]

**Pull request merged**